src/main/java/com/kalessil/phpStorm/phpInspectionsEA/inspectors/languageConstructions/BacktickOperatorUsageInspector.java

```diff
@@ -0,0 +1,96 @@
+package com.kalessil.phpStorm.phpInspectionsEA.inspectors.languageConstructions;
+
+import com.intellij.codeInspection.LocalQuickFix;
+import com.intellij.codeInspection.ProblemDescriptor;
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.PsiElementVisitor;
+import com.intellij.psi.SmartPointerManager;
+import com.intellij.psi.SmartPsiElementPointer;
+import com.jetbrains.php.lang.psi.PhpPsiElementFactory;
+import com.jetbrains.php.lang.psi.elements.FunctionReference;
+import com.jetbrains.php.lang.psi.elements.PhpShellCommandExpression;
+import com.jetbrains.php.util.PhpStringUtil;
+import com.kalessil.phpStorm.phpInspectionsEA.openApi.BasePhpElementVisitor;
+import com.kalessil.phpStorm.phpInspectionsEA.openApi.BasePhpInspection;
+
+import java.util.regex.Pattern;
+
+import org.jetbrains.annotations.NotNull;
+
+/*
+ * This file is part of the Php Inspections (EA Extended) package.
+ *
+ * (c) Vladimir Reznichenko <kalessil@gmail.com>
+ * (c) David Rodrigues <david.proweb@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+public class BacktickOperatorUsageInspector extends BasePhpInspection {
+    static final String message = "Prefer shell_exec() function instead of this backtick operator";
+
+    @NotNull
+    public String getShortName() {
+        return "BacktickOperatorUsageInspection";
+    }
+
+    @Override
+    @NotNull
+    public PsiElementVisitor buildVisitor(@NotNull final ProblemsHolder holder, final boolean isOnTheFly) {
+        return new BasePhpElementVisitor() {
+            @Override
+            public void visitPhpShellCommand(@NotNull final PhpShellCommandExpression shellCommandExpression) {
+                holder.registerProblem(shellCommandExpression, message, new UseShellExecQuickFix(shellCommandExpression));
+            }
+        };
+    }
+
+    private static class UseShellExecQuickFix implements LocalQuickFix {
```

Comment:
There is an easy way for this: com.kalessil.phpStorm.phpInspectionsEA.inspectors.apiUsage.IsNullFunctionUsageInspector.CompareToNullFix. Visitor will generate text for replacement and the fix will apply it (the base fixer is stable and bug-free as used globally).

Reply:
Oh, great. I will check that way!

Reply:
I am working on this PR right now and I checked your suggestion here, but it seems to be an excessive work to inspection if the fix is not applied (the inspection will generates the fix, allocate it, but could not be used by the user; and the fix will be regenerated on each code change). Am I right?

Reply:
Yes, you are right. A little bit of text manipulation and fixer creation is taking an extremely small amount of resources, in fact, I migrating to this approach last year.

Reply:
I think that it could be simplified without use additional resources (even than small) by using closures. Maybe we can implements that or you think better do like you suggested anyway?